cmd/compute-domain-controller/main.go

```diff
@@ -31,13 +31,16 @@ import (
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 	"github.com/urfave/cli/v2"
 
+	"k8s.io/component-base/logs"
 	"k8s.io/component-base/metrics/legacyregistry"
 	"k8s.io/klog/v2"
 
 	_ "k8s.io/component-base/metrics/prometheus/restclient" // for client metric registration
 	_ "k8s.io/component-base/metrics/prometheus/version"    // for version metric registration
 	_ "k8s.io/component-base/metrics/prometheus/workqueue"  // register work queues in the default legacy registry
 
+	"github.com/gdexlab/go-render/render"
```

Comment:
I get a 404 when trying to visit the github repo. Has it been removed?

Reply:
Yes, it's a bit of a fishy landscape.

Pointers:
- https://pkg.go.dev/github.com/gdexlab/go-render/render
- https://github.com/jgehrcke/k8s-dra-driver-gpu/blob/fc930fdd9658d32d8dee55a747903b0acdb33dc7/vendor/github.com/gdexlab/go-render/render/render.go (from this branch, notice the `AsCode()` method)

Appears to be old code from a Google/Chromium team. There are various variants out there.

- There's https://github.com/luci/go-render/blob/9a04cc21af0f2e328bfb81741ec9e8967c4f6541/render/render.go (no `AsCode()` method).
- There's a maintained version: https://github.com/smarty/assertions/blob/9be33929e2cfe616123c41d63712b12ca83f4184/internal/go-render/render/render.go

Reply:
Does:
```
			flagsJSON, _ := json.MarshalIndent(flags, "", "  ")
			klog.Infof("config: %v", flagsJSON)
```
or even:
```
			klog.Infof("config: %#v", flags)   
```
get us roughly to where we want to be? (We may need to make the members public for the rendering to work properly).

Reply:
Some quick testing:
```
	flagsJson, _ := json.MarshalIndent(f, "", "  ")
	fmt.Printf("flags json.MarshalIndent: %v\n", flagsJson)
	fmt.Printf("flags with #v: %#v\n", f)
```
outputs:
```flags json.MarshalIndent: [123 125]
flags with #v: main.Flags{kubeClientConfig:flags.KubeClientConfig{KubeConfig:"", KubeAPIQPS:0, KubeAPIBurst:0}, loggingConfig:(*flags.LoggingConfig)(nil), featureGateConfig:(*flags.FeatureGateConfig)(nil), podName:"", namespace:"", imageName:"", maxNodesPerIMEXDomain:0, httpEndpoint:"", metricsPath:"", profilePath:"", additionalNamespaces:cli.StringSlice{slice:[]string(nil), separator:cli.separatorSpec{sep:"", disabled:false, customized:false}, hasBeenSet:false, keepSpace:false}}```

Reply:
> get us roughly to where we want to be? 

Not really. I came from the JSON approach and looked for something better.

Compare that with https://github.com/NVIDIA/k8s-dra-driver-gpu/pull/633#discussion_r2391328040.

The logging config isn't shown, for example.

This readme describes rather well what go-render is doing: https://github.com/luci/go-render

>  Unlike the "%#v" format string, this resolves pointer types' contents in structs, maps, and slices/arrays and prints their field values

I see that potentially here we have a security concern (because clearly configuration is often user-given). But I also think that if we were to actually use `luci/go-render` we're good.